### DISCUSSION TITLE
PHP executable path in Adobe Brackets

### DISCUSSION THREAD

**Discussion by dmroliff** [brackets-cont/brackets#149] [Q&A]
I am running Adobe Brackets v 2.0

 on a 2017 iMac (Retina - 27 inch) with macOS Monterey 12.2

I want to point the path to PHP inside of MAMP. What is the correct json format and code to do this in Bracket's brackets.json file?

I've seen this code for someone using XAMP on a PC:

`"php": {
    "executablePath": "C:\\xampp\\php\\php.exe"
}`

I tried this in my brackets.json file:

`"php": {
    "executablePath": “/applications/MAMP/bin/php/php7.4.21/bin/“
}
`

but apparently that is not correct JSON formatting. I've tried using the double forward slashes to no avail.

It IS the correct path to the PHP 7.4.21 php executable file within my MAMP directory though.

This is what is currently in my brackets.json file:

`{
    "brackets-eslint.gutterMarks": true,
    "brackets-eslint.useLocalESLint": false,
    "fonts.fontSize": "12px",
    "fonts.fontFamily": "'SourceCodePro-Medium', ＭＳ ゴシック, 'MS Gothic', monospace",
    "externalApplications": {
        "jpg": "Adobe Photoshop 2021",
        "jpeg": "Adobe Photoshop 2021",
        "png": "Adobe Photoshop 2021",
        "svg": "Brackets",
        "psd": "Adobe Photoshop 2021",
        "ai": "Adobe Photoshop 2021"
    }
}`

I want to do this so that I can debug PHP code within Brackets.

Any suggestions?

Thanks.

Dennis

![brackets-preferences](https://user-images.githubusercontent.com/51270830/153541510-6e433447-467a-4543-8e80-3cc80b67af0d.jpg)
![PHP-runtime-not-found](https://user-images.githubusercontent.com/51270830/153541813-7b966b4d-6d4f-4a60-bc24-1aafca24ff37.jpg)

update:

I have added the [recommended JSON](https://github.com/adobe/brackets/wiki/PHP-Support-in-Brackets) to my Brackets preferences and am still getting the 'Error Reading Preferences' message. Here is what I have (lines 2 thru 4 are what I have added):

`{
"php": {
        "executablePath": "php" "/applications/MAMP/bin/php/php7.4.21/bin/php”	
}
    "brackets-eslint.gutterMarks": true,
    "brackets-eslint.useLocalESLint": false,
    "fonts.fontSize": "12px",
    "fonts.fontFamily": "'SourceCodePro-Medium', ＭＳ ゴシック, 'MS Gothic', monospace",
    "externalApplications": {
        "jpg": "Adobe Photoshop 2021",
        "jpeg": "Adobe Photoshop 2021",
        "png": "Adobe Photoshop 2021",
        "svg": "Brackets",
        "psd": "Adobe Photoshop 2021",
        "ai": "Adobe Photoshop 2021"
    }
}`

Can anyone see invalid JSON in lines 2 thru 4?

![valid-JSON](https://user-images.githubusercontent.com/51270830/153689513-76688022-a32e-407c-aa9b-8f13026dbb37.jpg)

**Comment by charlypa** [comment]
@dmroliff 
looks like your php path is not complete `"php": { "executablePath": “/applications/MAMP/bin/php/php7.4.21/bin/“ } `
Try replacing it with  `"php": { "executablePath": “/applications/MAMP/bin/php/php7.4.21/bin/php“ } `

Please check this document also for more information
https://github.com/adobe/brackets/wiki/PHP-Support-in-Brackets

**Reply by dmroliff** [reply]
Hi @charlypa ,
Thanks for the input. I tried replacing it with your suggestion and no joy.

I get the error message:
![error-reading-preferences](https://user-images.githubusercontent.com/51270830/153682978-bbbbfcae-7cb7-478c-b603-a05833278bb1.jpg)

I'll check out the link you provided.

Dennis

**Reply by charlypa** [reply]
@dmroliff 
There is a missing `,` after  PHP block. Once you add the missing comma `,` the color of the below text in `brackets.json` changes to `PHP.`

**Reply by dmroliff** [reply]
Hi @charlypa 

I added the comma but am still getting the 'Error Reading Preferences' message. Do I need double slashes in my executable path in order to 'escape'?

```{
"php": {
        "executablePath": "php" "/applications/MAMP/bin/php/php7.4.21/bin/php”	
},
    "brackets-eslint.gutterMarks": true,
    "brackets-eslint.useLocalESLint": false,
    "fonts.fontSize": "12px",
    "fonts.fontFamily": "'SourceCodePro-Medium', ＭＳ ゴシック, 'MS Gothic', monospace",
    "externalApplications": {
        "jpg": "Adobe Photoshop 2021",
        "jpeg": "Adobe Photoshop 2021",
        "png": "Adobe Photoshop 2021",
        "svg": "Brackets",
        "psd": "Adobe Photoshop 2021",
        "ai": "Adobe Photoshop 2021"
    }
   
}
```
![JSON-002](https://user-images.githubusercontent.com/51270830/153723897-d129888f-c02e-401a-ba15-1e42b5001f63.jpg)

**Reply by charlypa** [reply]
@dmroliff 
I think you have an extra `php` in the executable path and also you have an extra `}` at bottom

Try with this code to see if it works
```json

"php": {
        "executablePath":  "/applications/MAMP/bin/php/php7.4.21/bin/php”
},

"brackets-eslint.gutterMarks": true,
"brackets-eslint.useLocalESLint": false,
"fonts.fontSize": "12px",
"fonts.fontFamily": "'SourceCodePro-Medium', ＭＳ ゴシック, 'MS Gothic', monospace",
"externalApplications": {
    "jpg": "Adobe Photoshop 2021",
    "jpeg": "Adobe Photoshop 2021",
    "png": "Adobe Photoshop 2021",
    "svg": "Brackets",
    "psd": "Adobe Photoshop 2021",
    "ai": "Adobe Photoshop 2021"
}
   


```


If the above works please  try below also to get full PHP support


```json

"php": {
        "enablePhpTooling": true,
        "executablePath":  "/applications/MAMP/bin/php/php7.4.21/bin/php”,
        "memoryLimit": "4095M", 
        "validateOnType": "false"
         
},

"brackets-eslint.gutterMarks": true,
"brackets-eslint.useLocalESLint": false,
"fonts.fontSize": "12px",
"fonts.fontFamily": "'SourceCodePro-Medium', ＭＳ ゴシック, 'MS Gothic', monospace",
"externalApplications": {
    "jpg": "Adobe Photoshop 2021",
    "jpeg": "Adobe Photoshop 2021",
    "png": "Adobe Photoshop 2021",
    "svg": "Brackets",
    "psd": "Adobe Photoshop 2021",
    "ai": "Adobe Photoshop 2021"
}
   


```

**Reply by dmroliff** [reply]
> @dmroliff There is a missing `,` after PHP block. Once you add the missing comma `,` the color of the below text in `brackets.json` changes to `PHP.`

Hi @charlypa ,

This was the answer. Thanks. There were a couple missing commas. See reference jpeg below.
Let me know how to mark your suggestion as the correct answer. 😃

Regards,

Dennis

![correct-preferences-JSON](https://user-images.githubusercontent.com/51270830/153768027-c3152052-ec54-4123-9e79-1455ecce7367.jpg)

**Reply by charlypa** [reply]
@dmroliff 
The `JSON` file format in `adobe` documentation is not correct. We will see what we could see to update PHP documentation in brackets.
